src/blockchain/inherents.ts

```diff
@@ -106,7 +104,14 @@ export class SetValidationData implements CreateInherents {
       // chain started with genesis, mock 1st validationData
       newData = MOCK_VALIDATION_DATA
     } else {
-      const method = meta.registry.createType<GenericExtrinsic>('GenericExtrinsic', extrinsics[this.#expectedIndex])
+      const methods = extrinsics.map((extrinsic) =>
+        meta.registry.createType<GenericExtrinsic>('GenericExtrinsic', extrinsic)
+      )
+
+      const method = methods.find((method) => 'validationData' in method.args?.[0])
```

Comment:
I think it's best to perform decoding inside find to avoid decoding all extrinsic

Reply:
I hesitated, because find will return the method, not generated extrinsic, so it will require converting it again.